uint32 r
uint32 g
uint32 b
uint32 c